server/app/controllers/admin/AdminProgramStatusesController.java

```diff
@@ -34,6 +57,210 @@ public Result index(Http.Request request, long programId) throws ProgramNotFound
     if (!statusTrackingEnabled) {
       return notFound("status tracking is not enabled");
     }
-    return ok(statusesView.render(service.getProgramDefinition(programId)));
+    requestChecker.throwIfProgramNotDraft(programId);
+    ProgramDefinition program = service.getProgramDefinition(programId);
+    int previousStatusCount = program.statusDefinitions().getStatuses().size();
+
+    Optional<Form<ProgramStatusesEditForm>> maybeEditForm = Optional.empty();
+    if (request.method().equalsIgnoreCase(HttpVerbs.POST)) {
+      Form<ProgramStatusesEditForm> editForm;
+      Optional<StatusDefinitions> updatedStatus;
+      try {
+        Pair<Form<ProgramStatusesEditForm>, Optional<StatusDefinitions>> validated =
+            validateEditRequest(request, program);
+        editForm = validated.getLeft();
+        updatedStatus = validated.getRight();
+      } catch (MissingStatusException e) {
+        return redirect(routes.AdminProgramStatusesController.index(programId).url())
+            .flashing(
+                "error",
+                "The status being edited no longer exists and may have been modified in a separate"
```

Comment:
This will show as a toast?  Can we give the user something actionable to do to resolve the issue? EG reload the page.